### ISSUE TITLE
Suggestion: expand the scope extents

### ISSUE THREAD

**Issue by tlemo** [leodevbro/vscode-blockman#59]
Currently, the scope rectangle is fitting the non-whitespace content tightly, ex:

![image](https://user-images.githubusercontent.com/13824553/147841246-6ec92734-5c88-4721-bc64-6fdafc2530ef.png)

For most programming languages it may be easier to read if the scope are expanded:

1. Horizontally, to the outer right edge:

![image](https://user-images.githubusercontent.com/13824553/147841263-e5fd1c98-b225-4d2c-bc6c-75184fc6ac50.png)

2. Vertically, to include the scope brackets and ideally the parent statement (screenshot from Qt Creator)

![image](https://user-images.githubusercontent.com/13824553/147841293-2c70e448-b6f0-434d-911c-3c16911dbd44.png)

**Comment by leodevbro** [comment]
The question is: What should happen in this situation? Should the blocks be rendered in tight style or in expanded style? Don't you find the right side image less attractive? Yeah, I understand that such code structure will be very rare if not never, but the current Blockman style feels universally applicable to any kind of code structures. Personally for me, expended blocks feel less pretty and less readable. And less flexible.
![2022-01-01_13-35-38](https://user-images.githubusercontent.com/54504880/147848034-7e5a442a-b3da-46ff-b744-dfa291c2745f.png)

**Comment by tlemo** [comment]
This is indeed subjective, but prefer the expanded blocks. Perhaps it could be a user-configurable setting?

**Comment by leodevbro** [comment]
It does not seem an easy task, but I'll try to make it as an option in settings. And actually I want to try changing the simple div structure into svg structure, so it will be more flexible, I hope.

**Comment by leodevbro** [comment]
@tlemo, About the expanded blocks, I also want to say that the main philosophy of Blockman is that the rendered blocks should be as small as possible, so it will be easier for our eyes to focus. It just feels a good approach for me. So... You still prefer expanded blocks?

**Comment by tlemo** [comment]
Yes, my preference is still for the expanded blocks. (for reference, I find the the Qt Creator's approach best suited for coding)

It's just a suggestion though.

**Comment by leodevbro** [comment]
BTW, I see that Qt Creator's approach is different not only by expanded blocks, but also it renders parent blocks instead of child blocks. Blockman has child-block-rendering style.

I also want to implement parent-block-rendering style as an option for users, but it seems very difficult, because I think I will need all the tokenizers/parsers for all languages, because locating parent start/end points are much harder than locating child  start/end points.

Parent-block rendering seems difficult even in the advanced software like `Qt Creator`, for example, I found this bug. I know such kind of code formatting is very rare, if not never, but who knows, maybe some people like that way, so the rendering engine should work universally for any kind of formatting. The child-block-rendering style is so simple that I think it can work for absolutely any formatting.

![2022-01-10_12-39-25](https://user-images.githubusercontent.com/54504880/148739715-8e1cd6b5-3ab5-40ee-98d7-51709075c9d0.png)

**Comment by leodevbro** [comment]
> you can report the issue you discovered [here](https://bugreports.qt.io/)

yeah, I'll do it. Thanks for the link

> solution could be to expand the highlighted block to include the scope delimiters (curly brackets for example).

I'll think about it.

**Comment by leodevbro** [comment]
@tlemo, @ndgayan, I think I implemented the block-expanding settings.
The latest version of Blockman is 1.5.0 (2022-01-18).

![right-side-padding-strategies](https://user-images.githubusercontent.com/54504880/150006133-23702d55-27f3-47af-a3f7-e84284709eb8.gif)

**Comment by tlemo** [comment]
Looks great, thank you @leodevbro !
